src/Compilers/CSharp/Test/Semantic/Semantics/QueryTests.cs

```diff
@@ -2241,5 +2241,143 @@ public static Test Select<T>(System.Func<int, T> selector)
 
             CompileAndVerify(compilation, expectedOutput: "Select");
         }
+
+        [WorkItem(15910, "https://github.com/dotnet/roslyn/issues/15910")]
+        [Fact]
+        public void ExpressionVariablesInQueryClause_01()
+        {
+            var csSource = @"
+using System.Linq;
+
+class Program
+{
+    public static void Main(string[] args)
+    {
+        var a = new[] { 1, 2, 3, 4 };
+        var za = from x in M(a, out var q1) select x; // ok
+        var zc = from x in a from y in M(a, out var z) select x; // error 1
+        var zd = from x in a from int y in M(a, out var z) select x; // error 2
+        var ze = from x in a from y in M(a, out var z) where true select x; // error 3
+        var zf = from x in a from int y in M(a, out var z) where true select x; // error 4
+        var zg = from x in a let y = M(a, out var z) select x; // error 5
+        var zh = from x in a where M(x, out var z) == 1 select x; // error 6
+        var zi = from x in a join y in M(a, out var q2) on x equals y select x; // ok
+        var zj = from x in a join y in a on M(x, out var z) equals y select x; // error 7
+        var zk = from x in a join y in a on x equals M(y, out var z) select x; // error 8
+        var zl = from x in a orderby M(x, out var z) select x; // error 9
+        var zm = from x in a orderby x, M(x, out var z) select x; // error 10
+        var zn = from x in a group M(x, out var z) by x; // error 11
+        var zo = from x in a group x by M(x, out var z); // error 12
```

Comment:
Would you mind also including a deconstruction (declaration)? I don't think we've covered it in LINQ clauses elsewhere and it may have become parseable after recent changes.